### PR TITLE
Add Prettier with tslint plugin and precommit hook

### PR DESCRIPTION
**Pull request by agubler**
**Type:** feature

The following has been addressed in the PR:

* [ ] There is a related issue
* [x] All code matches the [style guide](https://github.com/dojo/meta/blob/master/STYLE.md)
* [ ] Unit or Functional tests are included in the PR

**Description:**

Adds support for `Prettier` to for code style rules and formatting. Include precommit hook that runs prettier on all `.ts` files and an npm script that runs prettier against the codebase.

Related to https://github.com/dojo/meta/issues/206

### CODE REVIEW

File: README.md [hunk 1]
```diff
@@ -401,7 +401,17 @@ formatUnit(1000, 'meter', null, 'fr); // 1 000 mètres'
 ## How do I contribute?
 
 We appreciate your interest!  Please see the [Guidelines Repository](https://github.com/dojo/guidelines#readme) for the
-Contributing Guidelines and Style Guide.
+Contributing Guidelines.
+
+### Code Style
```
Comment (agubler) [inline README.md]: Be good to get a specific review on the wording here.

File: .github/PULL_REQUEST_TEMPLATE.md [hunk 2]
```diff
@@ -3,7 +3,7 @@
 The following has been addressed in the PR:
 
 * [ ] There is a related issue
-* [ ] All code matches the [style guide](https://github.com/dojo/meta/blob/master/STYLE.md)
+* [ ] All code has been formatted with `prettier`
```
Comment (agubler) [inline .github/PULL_REQUEST_TEMPLATE.md]: Is this okay for the pull request template?

Comment (mwistrand) [reply .github/PULL_REQUEST_TEMPLATE.md]: I think this is fine.

Comment (tomdye) [reply .github/PULL_REQUEST_TEMPLATE.md]: sounds ok to me. Unless you replace `with` with `via` ?

File: README.md (hunk 1 above)
Comment (tomdye) [reply README.md]: reads well to me

**Comment by agubler** [comment]
Closing will re-open instead of trying to resolve all these conflicts.

**Pull request closed without merging**